test/Microsoft.AspNetCore.Mvc.TagHelpers.Test/CacheTagHelperTest.cs

```diff
@@ -648,6 +648,67 @@ public void UpdateCacheEntryOptions_SetsCachePreservationPriority()
             Assert.Equal(2, calls);
         }
 
+        [Fact]
+        public async Task ProcessAsync_ExceptionInProcessing_DoNotThrowInSubsequentRequests()
+        {
+            // Arrange
+            var id = "unique-id";
+            var childContent = "some-content";
+            var cache = new MemoryCache(new MemoryCacheOptions());
+
+            var tagHelperContext1 = GetTagHelperContext(id);
+            var tagHelperContext2 = GetTagHelperContext(id);
+
+            var tagHelperOutput1 = new TagHelperOutput(
```

Comment:
@sebastienros The problem likely not in caching faulted task.

In current version of dev (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs#L107) setting of entry options and value occurs after awaiting of output.

When output is faulted no entry value is set up. Entry also is not beeing invalidated (https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs#L116) because no options with cancel token was set up.

Then in subsequent requests we get entry, get its value (that is actually null) and trying to await it getting NullPointerException here https://github.com/aspnet/Mvc/blob/dev/src/Microsoft.AspNetCore.Mvc.TagHelpers/CacheTagHelper.cs#L133